### ISSUE TITLE
Add support for structured authentication configuration

### ISSUE THREAD

**Issue by bmiddha** [siderolabs/talos#8933]
## Feature Request

Add support for structured authentication configuration to add authentication providers to the cluster.

### Description

I'm looking into using the Structured Authentication Configuration feature in my Talos cluster. The `APIServerConfig` does not support adding this config file for `kube-apiserver`.

Feature gate

| Feature | Default | Stage | Since | Until |
| --- | --- | --- | --- | --- |
| StructuredAuthenticationConfiguration |false | Alpha | 1.29 | 1.29 |
| StructuredAuthenticationConfiguration |true | Beta | 1.30 | - |

https://kubernetes.io/docs/reference/command-line-tools-reference/feature-gates/

It requires a file provided as `--authentication-config` to `kube-apiserver`.

> To use structured authentication configuration, you specify the path to the authentication configuration using the --authentication-config command line argument in the API server. 

https://kubernetes.io/blog/2024/04/25/structured-authentication-moves-to-beta/

**Comment by bmiddha** [comment]
Is this something that I can contribute? I can follow the pattern used for the `--admission-control-config-file` and `--audit-policy-file` params.

**Comment by frezbo** [comment]
> Is this something that I can contribute? I can follow the pattern used for the `--admission-control-config-file` and `--audit-policy-file` params.

Yes, this would be good. The only minor concern is the n-5 support Talos has for k8s, so this can only support for clusters running 1.30, probably can handle that in https://github.com/siderolabs/go-kubernetes/

**Comment by smira** [comment]
You should be able to do a workaround by writing the config file via `.machine.files` somewhere under `/var` and then mounting that up into the apiserver static pod via hostPath volume mounts. Not beautiful, but is should be possible.

**Comment by hegerdes** [comment]
Just some extra context:

The `StructuredAuthenticationConfiguration` gust upgraded vom `v1beta1` to `v1` and the `v1` will likely be GA in k8s `v1.34`. See [131752](https://github.com/kubernetes/kubernetes/pull/131752) 

Its feature is also requested in #10905 and #11065. 

I'm also using the workaround mentioned be smira, but first party support would be great. A dedicated config docs might be overkill form my perspective, but having a property called `authenticationConf` (or similar) and let this have the native k8s `StructuredAuthenticationConfiguration` would avoid dublicatin the conf structure.  
Supporting different versions would requires a little extra care, I but since `v1.30` we can assume it is enabled and I thing `v1beta1` and `v1` have the same structure - haven't checked yet though

**Comment by michaelbeaumont** [comment]
Is it expected to be impossible to unset apiserver flags like `--anonymous-auth`? I can't figure out a way to get rid of it, [when using `--authentication-config` and `anonymous` it's causing the apiserver not to start](https://kubernetes.io/docs/reference/access-authn-authz/authentication/#anonymous-authenticator-configuration).

**Comment by clementnuss** [comment]
@smira once a decision has been made regarding K8s config documents, I'm happy to implement this issue as I'll also need it. similarly to how I did it here: https://github.com/siderolabs/talos/pull/11641

**Comment by michaelbeaumont** [comment]
🙏 not stale
